Support SPI communication

afontaine-invn opened the pull request:
Add WORKAROUND in driver (0x4 in DRIVE_CONFIG3)
Fix SPI read/write with multiple bytes
Move CS pin to D8

Testing done : 
- Run the 3 examples in SPI mode with different config options.

rbuisson-invn commented on drivers/sensor/icm42670S/imu/inv_imu_driver.c:
```diff
@@ -1483,6 +1483,10 @@ static int configure_serial_interface(inv_imu_device_t *s)
 			return INV_ERROR_BAD_ARG; /* Not supported */
 		status |= inv_imu_write_reg(s, DEVICE_CONFIG, 1, &value);
 
+		/* Workaround : Overwrite DRIVE_CONFIG3 register with 0x4 value */
```
Do we want to emphasize publicly that the IMU has a bug?

afontaine-invn replied:
Could we address this point prior to open the PR to zephyr? I would merge it as it fixes the SPI functionality. This way we could move forward with more testing/formatting/...

sriccardi-invn replied:
We cannot fix this now because we have still discussin with HW Team.
We must be sure that this is the right path.
Then this code is part of emd driver. It must be fixed at this level officially.

afontaine-invn replied:
Ok, I removed the workaround

rbuisson-invn reviewed (changes requested):
Ok but need to clarify if we want to show there is a bug or not

rbuisson-invn commented on drivers/sensor/icm42670S/icm42670S_spi.c:
```diff
@@ -21,46 +21,40 @@ static int icm42670S_bus_check_spi(const union icm42670S_bus *bus)
+	uint8_t cmd[] = {reg & 0x7F, *buf};
 	const struct spi_buf tx_buf = {.buf = cmd, .len = sizeof(cmd)};
-	const struct spi_buf_set tx = {.buffers = &tx_buf, .count = 1};
+	const struct spi_buf_set tx = {.buffers = &tx_buf, .count = size};
```
I'm pretty sure ".count=size" should be replaced by ".count=1".
It probably works because buf length is always 1, but the behavior won't be correct if the driver tries to write 2 bytes!

afontaine-invn replied:
Ok, I modified this.

Commit:
Support SPI communication

Add WORKAROUND in driver (0x4 in DRIVE_CONFIG3)
Fix SPI read/write with multiple bytes

Commit:
Fix write burst access

Remove workaround
